crates/astria-sequencer/src/checked_actions/bridge/bridge_unlock.rs

```diff
@@ -0,0 +1,649 @@
+        // `BridgeUnlock` parsing.
+        ensure!(action.amount > 0, "amount must be greater than zero");
+        ensure!(
+            action.memo.len() <= 64,
+            "memo must not be more than 64 bytes"
+        );
+        ensure!(
+            !action.rollup_withdrawal_event_id.is_empty(),
+            "rollup withdrawal event id must be non-empty",
+        );
+        ensure!(
+            action.rollup_withdrawal_event_id.len() <= 256,
+            "rollup withdrawal event id must not be more than 256 bytes",
+        );
+        ensure!(
+            action.rollup_block_number > 0,
+            "rollup block number must be greater than zero",
+        );
+
+        state
+            .ensure_base_prefix(&action.to)
+            .await
+            .wrap_err("destination address has an unsupported prefix")?;
+        state
+            .ensure_base_prefix(&action.bridge_address)
+            .await
+            .wrap_err("source address has an unsupported prefix")?;
+
+        let bridge_account_ibc_asset = state
+            .get_bridge_account_ibc_asset(&action.bridge_address)
+            .await
+            .wrap_err("failed to get bridge account asset ID; account is not a bridge account")?;
+
+        let checked_action = Self {
+            action,
+            tx_signer: tx_signer.into(),
+            bridge_account_ibc_asset,
+        };
+        checked_action.run_mutable_checks(state).await?;
+
+        Ok(checked_action)
+    }
+
+    #[instrument(skip_all, err(level = Level::DEBUG))]
+    pub(in crate::checked_actions) async fn run_mutable_checks<S: StateRead>(
+        &self,
+        state: S,
+    ) -> Result<()> {
+        if PURE_UNLOCK
+            && state
+                .is_a_bridge_account(&self.action.to)
+                .await
+                .wrap_err("failed to check if `to` address is a bridge account")?
+        {
+            bail!("bridge accounts cannot receive bridge unlocks");
+        }
+
+        let withdrawer = state
+            .get_bridge_account_withdrawer_address(&self.action.bridge_address)
+            .await
+            .wrap_err("failed to get bridge account withdrawer address")?
+            .ok_or_eyre("bridge account must have a withdrawer address set")?;
+        ensure!(
+            *self.tx_signer.as_bytes() == withdrawer,
+            "signer is not the authorized withdrawer for the bridge account",
+        );
+
+        if let Some(existing_block_num) = state
+            .get_withdrawal_event_block_for_bridge_account(
+                &self.action.bridge_address,
+                &self.action.rollup_withdrawal_event_id,
+            )
+            .await
+            .wrap_err("failed to read withdrawal event block number from storage")?
+        {
+            bail!(
+                "withdrawal event ID `{}` used by block number {existing_block_num}",
+                self.action.rollup_withdrawal_event_id
+            );
```

Comment:
```suggestion
            bail!(
                "withdrawal event ID `{}` was already executed (rollup block number {existing_block_num})",
                self.action.rollup_withdrawal_event_id
            );
```

Reply:
Done in [73ec272](https://github.com/astriaorg/astria/pull/2142/commits/73ec27210915e8a170768ff722a20d45297b4f81).